Quantum_Basics.jl

```diff
@@ -0,0 +1,1888 @@
+
+We can further reduce the dimensionality of the problem by noting that the normalisation to length 1 of the vector removes one more degree of freedom. Therefore, only two numbers are sufficient to characterize the state: 
+ - The complex phase between the states
+ - How much the state points in the z-direction"
+
+# ╔═╡ f6b42b7d-c614-4c95-a105-d7dced988825
+function parametrize_Bloch_sphere(state)
+	return [2*acos(abs(state[1])),angle(state[1]) - angle(state[2])]
+end
+
+# ╔═╡ f802e2f0-201e-490b-937c-82e25bb640e0
+function Bloch_point(state)
+	parameters = parametrize_Bloch_sphere(state)
+	return Float64[sin(parameters[1])*cos(parameters[2]), sin(parameters[1])*sin(parameters[2]), cos(parameters[1])]
+end
+
+# ╔═╡ 70539c8d-7565-4554-b60e-22220151b38b
+time_range = 0:0.1:10
+
+# ╔═╡ f0716eed-8be2-4031-9345-c28386fc1412
+initial_state = [1,0]
+
+# ╔═╡ 4a5aa91c-cd22-481c-9e99-bf24a1eaa092
+@bind time PlutoUI.Slider(time_range)
+
+# ╔═╡ a0c4484b-0869-427b-9885-de3072bc8fb4
+md"Because the point can be a bit difficult to follow, we also plot the trace. Write the name of the function, which returns the time-evolution operator here:"
+
+# ╔═╡ a6269134-0ca9-464a-b05e-e4a17c1eac46
+time_evol = time_evolution_B_x
+
+# ╔═╡ 67988f84-dc78-4c9d-bc6b-260fd36556a6
+time_evolved_state = time_evol(time)*initial_state
+
+# ╔═╡ 28728934-b4b5-4b58-b44a-2ea9c5f51f5b
+plotted_point = Bloch_point(time_evolved_state)
+
+# ╔═╡ ca85162f-4b9a-4d12-89f1-2126321e4071
+let # Define the sphere's radius and parameterize the surface using spherical coordinates
+radius = 1
+u = range(0, stop=2π, length=50)
+v = range(0, stop=π, length=50)
+x = [radius * sin(v[j]) * cos(u[i]) for i in 1:length(u), j in 1:length(v)]
+y = [radius * sin(v[j]) * sin(u[i]) for i in 1:length(u), j in 1:length(v)]
+z = [radius * cos(v[j]) for i in 1:length(u), j in 1:length(v)]
+
+plotted_points = zeros(Float64, (3,length(time_range)))
+
+for i in 1:length(time_range)
+	plotted_points[:,i] = Bloch_point(time_evol(time_range[i])*initial_state)
+end
+	
+# Plot the semi-transparent sphere
+fig, ax, plot = surface(x, y, z, color=:blue, transparency=true, alpha=0.3)
+
+lines!(ax, plotted_points[1,:], plotted_points[2,:], plotted_points[3,:])
+
+scatter!(ax, [plotted_point[1]], [plotted_point[2]], [plotted_point[3]], color = :red)
+# Display the figure
+fig
+end
+
+# ╔═╡ 4b1d02a0-7a85-40fc-8b8d-b8c753a68dc2
+md"Finally, we want to do the same thing for a general Hamiltonian, which may also be time dependant. For this we define a function Hamiltonian, which returns the matrix at any particular time."
+
+# ╔═╡ f0201489-9e3c-462e-8978-3c47ed7fa569
+function Hamiltonian_t(t)
+	return [sin(t) cos(t); cos(t) -2sin(t)]
+end
+
+# ╔═╡ 72d32760-3c54-406b-b59a-1d4a8f087488
+function Generic_Time_Evolution(time)#Write the function which returns the time evolution operator U(t) for a generic Hamiltonian. Assume ħ=1.
+	return cis(Hamiltonian_t(time)*time)
+end
```

Comment:
I wonder how you'd best do the integration. Maybe [FFTW.jl](https://github.com/JuliaMath/FFTW.jl) would be best, since you can do the Fourier transform once, and then use it to evaluate the integral at different points.

If that's too much work for you, consider Gauss or Clenshaw-Curtis weights and nodes:
- [FastGaussQuadrature.jl](https://github.com/JuliaApproximation/FastGaussQuadrature.jl) for Gauss quadrature.
- [FastTransforms.jl](https://juliaapproximation.github.io/FastTransforms.jl/dev/#FastTransforms.clenshawcurtisnodes) for Clenshaw Curtis.

If all else fails, I guess we're left with trapezoid or Simpon's rule or something like that.